widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java

```diff
@@ -212,20 +215,32 @@ public void startChat(
             String contextUrl,
             boolean useOverlays,
             Bundle savedInstanceState) {
+        Activity activity = Utils.getActivity(this.getContext());
         if (controller != null) {
-            Activity activity = Utils.getActivity(this.getContext());
-            controller.initChat(
-                    companyName,
-                    queueId,
-                    contextUrl,
-                    activity instanceof ChatActivity,
-                    activity instanceof ChatActivity && useOverlays,
-                    activity instanceof ChatActivity && savedInstanceState != null,
-                    this.theme,
+
+            controller.initChat(companyName, queueId, contextUrl);
+
+            controller.updatePermissions(
                     Settings.canDrawOverlays(this.getContext()),
                     NotificationManager.areNotificationsEnabled(this.getContext(), NotificationFactory.NOTIFICATION_CALL_CHANNEL_ID),
                     NotificationManager.areNotificationsEnabled(this.getContext(), NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID)
             );
+
+            if (activity instanceof ChatActivity && savedInstanceState == null) {
+                if (chatHeadsController != null) {
```

Comment:
seems like unnecessary nested if

Reply:
Will look into it while looking into chat bubble issues